example_12/doc/userdoc.rst

```diff
@@ -0,0 +1,168 @@
+
+The example shows how to write a simple chainable controller, and how to integrate it properly to have a functional controller chaining.
+
+For *example_12*, we will use RRBot, or ''Revolute-Revolute Manipulator Robot'', is a simple 3-linkage, 2-joint arm to demonstrate the controller chaining functionality in ROS2 control.
+
+For *example_12*, a simple chainable ros2_controller has been implemented that takes a vector of interfaces as an input and simple forwards them without any changes. Such a controller is simple known as a ``passthrough_controller``.
+
+.. include:: ../../doc/run_from_docker.rst
+
+Tutorial steps
+--------------------------
+
+1. To check that *RRBot* descriptions are working properly use following launch commands
+
+   .. code-block:: shell
+
+    ros2 launch ros2_control_demo_example_12 view_robot.launch.py
+
+   .. image:: rrbot.png
+    :width: 400
+    :alt: Revolute-Revolute Manipulator Robot
+
+   The ``joint_state_publisher_gui`` provides a GUI to change the configuration for *RRbot*. It is immediately displayed in *RViz*.
+
+
+2. To start *RRBot* with the hardware interface, open a terminal, source your ROS2-workspace and execute its launch file with
+
+   .. code-block:: shell
+
+    ros2 launch ros2_control_demo_example_12 rrbot.launch.py
+
+   The launch file loads and starts the robot hardware, controllers and opens RViz. In starting terminal you will see a lot of output from the hardware implementation showing its internal states. It uses an identical hardware interface as already discussed with *example_1*, see its docs on details on the hardware interface.
+
+   If you can see two orange and one yellow rectangle in in RViz everything has started properly. Still, to be sure, let’s introspect the control system before moving RRBot.
+
+3. Check if controllers are running by
+
+   .. code-block:: shell
+
+    ros2 control list_controllers
+
+   .. code-block:: shell
+
+    joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] active
+    joint2_position_controller[passthrough_controller/PassthroughController] active
+    joint1_position_controller[passthrough_controller/PassthroughController] active
+
+4. Check if the hardware interface loaded properly, by opening another terminal and executing
+
+   .. code-block:: shell
+
+    ros2 control list_hardware_interfaces
+
+   The output should be something like this:
+
+   .. code-block:: shell
+
+    command interfaces
+          joint1/position [available] [claimed]
+          joint1_position_controller/joint1/position [unavailable] [unclaimed]
+          joint2/position [available] [claimed]
+          joint2_position_controller/joint2/position [unavailable] [unclaimed]
+    state interfaces
+          joint1/position
+          joint2/position
+
+   At this stage the reference interfaces of controllers are listed under ``command_interfaces`` when ``ros2 control list_hardware_interfaces`` command is executed.
```

Comment:
Could you please add a description of the meaning of available and unavailable here? There is not a single word in the docs of ros2_control about that, we should also add something there: https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/controller_chaining.rst

Reply:
Sure, I can add that

Reply:
Yes, please add a PR for the docs of the CM :)